packages/node/src/integrations/hapi/index.ts

```diff
@@ -0,0 +1,169 @@
+import { captureException, configureScope, continueTrace, getActiveTransaction, startTransaction } from '@sentry/core';
+import type { Integration } from '@sentry/types';
+import { addExceptionMechanism, dynamicSamplingContextToSentryBaggageHeader, fill } from '@sentry/utils';
+
+import type { Boom, RequestEvent, ResponseObject, Server } from './types';
+
+function isResponseObject(response: ResponseObject | Boom): response is ResponseObject {
+  return response && (response as ResponseObject).statusCode !== undefined;
+}
+
+function isBoomObject(response: ResponseObject | Boom): response is Boom {
+  return response && (response as Boom).isBoom !== undefined;
+}
+
+function isErrorEvent(event: RequestEvent): event is RequestEvent {
+  return event && (event as RequestEvent).error !== undefined;
+}
+
+function sendErrorToSentry(errorData: object): void {
+  captureException(errorData, scope => {
```

Comment:
Note to self: if we merge https://github.com/getsentry/sentry-javascript/pull/9590 first, update this later!

Reply:
OK, merged this in yet, so we can refactor this to this instead:

```js
captureException(errorData, { 
  mechanism: {
        type: 'hapi',
        handled: false,
        data: {
          function: 'hapiErrorPlugin',
        },
      }
});
```